x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -772,6 +782,63 @@ private Expression simplify(BinaryComparison bc) {
         }
     }
 
+    /**
+     * Any numeric aggregates (avg, min, max, sum) acting on literals are converted to an iif(count(1)=0, null, literal*count(1)) for sum,
+     * and to iif(count(1)=0,null,literal) for the other three.
+     */
+    private static class ReplaceAggregatesOfLiterals extends OptimizerRule<LogicalPlan> {
+
+        @Override
+        protected LogicalPlan rule(LogicalPlan p) {
+            Holder<Boolean> isLocalRelation = new Holder<>(false);
+            p.forEachUp(a -> {
+                if (isLocalRelation.get() == false && a instanceof LocalRelation) {
+                    isLocalRelation.set(true);
+                }
+            });
+            
+            // there is no need for a IIF(COUNT(1)=0,NULL,literal) for local relations, since the aggregate will be folded further down
+            if (isLocalRelation.get()) {
+                return p;
+            }
+
+            Expression countOne = new Count(EMPTY, new Literal(Source.EMPTY, 1, DataTypes.LONG), false);
```

Comment:
nit: To keep the result data type consistent we could move those 2 lines inside the trasformation as:
```
 Expression countOne = new Cast(EMPTY, new Count(EMPTY, new Literal(Source.EMPTY, 1, fna.dataType()), false), fna.dataType());
                        Equals countEqZero = new Equals(EMPTY, countOne, new Literal(Source.EMPTY, 0, fna.dataType()));
```